### PR TITLE
Improved accessibility and Fixed main nav active states 

### PR DESCRIPTION
**Pull request by SameerJadav**
I have changed the docs site to make it more accessible and improve SEO. I also fixed the issue where the main navigation links were not showing in their active state. Here are the details of the changes we made:

- Added active states to main navigation links using [`useSelectedLayoutSegment()`](https://nextjs.org/docs/app/api-reference/functions/use-selected-layout-segment)
```ts
function isActive(href: string) {
  const segment = useSelectedLayoutSegment();
  if (!segment) return false;
  return href.startsWith(`/${segment}`);
}
```
- Changed muted colors a little to improve contrast ratio and gave buttons accessible names, increasing accessibility scores on the homepage from **85** to **100**
- Added `hoverOnlyWhenSupported`, which will prevent any hover styles from being shipped to devices that don't support hover (phones). You can look at the PR this feature was introduced in https://github.com/tailwindlabs/tailwindcss/pull/8394
- Added necessary meta tags for better SEO

We can also use 'next/images' in the docs and get all the benefits like lazy loading, blur placeholder, etc. and increase the performance even more. However, implementing this would require downloading each image.

Take your time to review the changes. I'm totally up for any feedback or suggestions you might have.

### CODE REVIEW

File: docs/src/components/main-nav.tsx [hunk 1]
```diff
@@ -16,31 +16,31 @@ export interface NavItem {
   label?: string;
 }
 
-export function MainNav(props: { items: NavItem[] }) {
-  const pathname = usePathname();
+function isActive(href: string) {
+  const segment = useSelectedLayoutSegment();
```
Comment (juliusmarminge) [inline docs/src/components/main-nav.tsx:20]: this doesn't adhere to rules of hooks

**Review by juliusmarminge — CHANGES_REQUESTED** [review]
I like the stylistic changes apart from the theme toggle item whose spacing have decreased:

old:
CleanShot 2023-08-16 at 18 05 35@2x

new:
CleanShot 2023-08-16 at 18 05 32@2x

File: docs/src/components/main-nav.tsx (hunk 1 above)
Comment (SameerJadav) [reply docs/src/components/main-nav.tsx:20]: i have moved it inside the component
```tsx
export function MainNav() {
  const isActive = (href: string) => {
    const segment = useSelectedLayoutSegment();
    if (!segment) return false;
    return href.startsWith(`/${segment}`);
  };
// ...
}
```
is this okay??

File: docs/next.config.mjs [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { getHighlighter } from "shiki";
 
 /** @type {import('next').NextConfig} */
 const nextConfig = {
+  swcMinify: true,
```
[the remark this answers is not in the document]
Comment (SameerJadav) [reply docs/next.config.mjs:10]: yes, it is the default js compiler. i'll remove it from the config

**Comment by SameerJadav** [comment]
> I like the stylistic changes apart from the theme toggle item whose spacing have decreased:
> 
> old: CleanShot 2023-08-16 at 18 05 35@2x
> 
> new: CleanShot 2023-08-16 at 18 05 32@2x

i've added back the spacing

File: docs/src/components/main-nav.tsx [hunk 3]
```diff
@@ -17,30 +17,31 @@ export interface NavItem {
...
-          )}
-        </nav>
-      ) : null}
-    </div>
+    <nav className="hidden md:flex md:items-center md:space-x-6">
+      {props.items?.map(
+        (item, index) =>
+          item.href && (
+            <Link
+              key={index}
```
Comment (juliusmarminge) [inline docs/src/components/main-nav.tsx:33]: key by href
```suggestion
              key={item.href}
```

**Review by juliusmarminge — APPROVED** [review]

File: docs/src/components/main-nav.tsx [hunk 4]
```diff
@@ -17,30 +17,31 @@ export interface NavItem {
...
-                  {item.title}
-                </Link>
-              )
-          )}
-        </nav>
-      ) : null}
-    </div>
+    <nav className="hidden md:flex md:items-center md:space-x-6">
+      {props.items?.map(
+        (item, index) =>
```
Comment (juliusmarminge) [inline docs/src/components/main-nav.tsx:30]:
```suggestion
        (item) =>
```

**Pull request merged**